### PR TITLE
CDAP-6595 Authorization for Metadata

### PR DESCRIPTION
**Pull request by rohitsinha54**
Opened PR https://github.com/caskdata/cdap/pull/7304 against develop. The PR was already reviewed there. 

Issue: https://issues.cask.co/browse/CDAP-6595

- Added authorization for Metadata (Tags and Properties)

Build: http://builds.cask.co/browse/CDAP-DUT5246-1
ITM: http://builds.cask.co/browse/CDAP-ITM-72

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataAdmin.java [hunk 1]
```diff
@@ -63,54 +66,67 @@ void addTags(NamespacedEntityId namespacedEntityId, String... tags)
...
 
   /**
    * Removes all the metadata (including properties and tags) for the specified {@link NamespacedEntityId}. This
    * API only supports removing metadata in {@link MetadataScope#USER}.
    *
    * @param namespacedEntityId the {@link NamespacedEntityId} to remove metadata for
    * @throws NotFoundException if the specified entity was not found
+   * @throws UnauthorizedException if the requesting user is does not have privileges to get tags for the given
    */
   void removeMetadata(NamespacedEntityId namespacedEntityId) throws NotFoundException;
```
Comment (Nishith) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataAdmin.java:131]: Shouldn't this and the three after this throw the UnauthorizedException?

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java [hunk 2]
```diff
@@ -596,7 +610,7 @@ public void getArtifactTags(HttpRequest request, HttpResponder responder,
                               @PathParam("artifact-name") String artifactName,
                               @PathParam("artifact-version") String artifactVersionStr,
                               @QueryParam("scope") String scope)
-    throws BadRequestException, NotFoundException {
+    throws Exception {
```
Comment (Nishith) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java:613]: What else does this throw, other than the three that the one's before it are throwing?

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java [hunk 3]
```diff
@@ -608,7 +622,7 @@ public void getProgramTags(HttpRequest request, HttpResponder responder,
                              @PathParam("app-id") String appId,
                              @PathParam("program-type") String programType,
                              @PathParam("program-id") String programId,
-                             @QueryParam("scope") String scope) throws NotFoundException, BadRequestException {
+                             @QueryParam("scope") String scope) throws Exception {
```
Comment (Nishith) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java:625]: Is this now throwing any additional exception other than UnauthorizedException?

**Review by Nishith — CHANGES_REQUESTED** [review]

**Comment by rohitsinha54** [comment]
On hold to focus on higher priority tasks of 4.1

**Pull request closed without merging**